overlord/configstate/applyconfighandler.go

```diff
@@ -19,7 +19,15 @@
 
 package configstate
 
-import "github.com/snapcore/snapd/overlord/hookstate"
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/hookstate"
+)
+
+// CachedTransaction is the index into the context cache where the initialized
+// transaction is stored.
+type CachedTransaction struct{}
```

Comment:
This is probably a better building block, and voids exposing the empty struct trick as a public API:
```
func ContextTransaction(context *hookstate.Context) *Transaction {
        t, ok := context.Cached(cachedTransaction{}).(*Transaction)
        if ok {
                return t
        }
        t := NewTransaction(context.State())
        context.Cache(cachedTransaction{}, t)
        return t
}
```
Note it already relies on the changes proposed in #2014.